src/main/kotlin/io/perfometer/dsl/HttpDsl.kt

```diff
@@ -111,18 +104,27 @@ class Scenario(
         private val baseURL: URL,
         private val builder: suspend HttpDsl.() -> Unit,
 ) {
+    private val consumerSelector: (Output) -> (ScenarioSummary) -> Unit = { getStatisticsConsumer(it) }
+
     private var runner: ScenarioRunner = CoroutinesScenarioRunner(KtorHttpClient())
 
     fun runner(runner: ScenarioRunner): Scenario {
         this.runner = runner
         return this
     }
 
-    fun run(userCount: Int, duration: Duration): ScenarioSummary {
-        return runner.runUsers(userCount, duration) {
-            builder(HttpDsl(baseURL, runner))
-        }.also {
-            StdOutStatisticsPrinter().print(it)
+    fun run(userCount: Int,
+            duration: Duration,
+            vararg outputTo: Output = arrayOf(Output.STDOUT)): ScenarioSummary {
+        return runner
+                .runUsers(userCount, duration) { builder(HttpDsl(baseURL, runner)) }
+                .also { consumeStatistics(it, outputTo.toList()) }
+    }
+
+    private fun consumeStatistics(summary: ScenarioSummary,
+                                  outputs: List<Output>) {
+        for (outputType in outputs) {
+            consumerSelector(outputType)(summary)
```

Comment:
I think all the logic of selecting consumer for output would be better suited in statistics package.
Maybe just move functions consumeStatistics and consumerSelector there?